includes/polling/os.inc.php

```diff
@@ -35,6 +35,13 @@
     set_device_location($location, $device, $update_array);
 }
 
+if ($name != $device['name']) {
```

Comment:
This is the wrong check and will always update.
Also the variable should probably be `$sysName`